StreamChatClient_v3/WebSocketClient/ConnectionState.swift

```diff
@@ -0,0 +1,62 @@
+//
+// ConnectionState.swift
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public typealias ConnectionId = String
+
+/// A web socket connection state.
+public enum ConnectionState: Equatable {
+  /// Provides additional information about the source of disconnecting.
+  public enum DisconnectionSource: Equatable {
+    /// A user initiated web socket disconnecting.
+    case userInitiated
+
+    /// A server initiated web socket disconnecting, an optional error object is provided.
+    case serverInitiated(error: ClientError? = nil)
+
+    /// The system initiated web socket disconnecting.
+    case systemInitiated
+  }
+
+  /// The web socket is not connected.
+  case notConnected
+
+  /// The web socket is connecting
+  case connecting
+
+  /// The web socket is connected, waiting for the connection id
+  case waitingForConnectionId
+
+  /// The web socket was connected.
+  case connected(connectionId: ConnectionId)
+
+  /// The web socket is disconnecting. `source` contains more info about the source of the event.
+  case disconnecting(source: DisconnectionSource)
+
+  /// The web socket is disconnected, optionally with an error. No reconnection attempts scheduled.
+  case disconnected(error: ClientError? = nil)
+
+  /// The web socket is waiting for reconnecting. Optinally, an error is provided with the reason why it was disconnected.
+  case waitingForReconnect(error: ClientError? = nil)
+
+  /// Checks if the connection state is connected.
+  public var isConnected: Bool {
+    if case .connected = self {
+      return true
+    }
+    return false
+  }
+
+  /// Checks if the connection state is currently in the process of connecting/disconnecting/reconnecting.
+  public var isActiveConnection: Bool {
```

Comment:
Comment and variable name doesn't match, what does this exactly do?

Reply:
It checks if something is happening with the connection, aka it's still somehow in an active state. I use it in the next PR to determine whether we need to start a background task.

Reply:
Then I suggest updating the comment to include the remaining states as well

Reply:
I'm ok with the name, but I can suggest the state `.isInProgress`.

Reply:
Because we unified `notConnected` and `disconnected` this is actually no longer needed :)